### PR TITLE
Roles

### PR DESCRIPTION
**Pull request by chriskrogh**
Fat PR

### CODE REVIEW

File: src/modules/data/useData.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useFirestore, useFirestoreCollectionData } from 'reactfire';
+
+const useData = <T, D>(COLLECTION_NAME: string) => {
```
Comment (jtaaa) [inline src/modules/data/useData.ts]: How about `useCollectionData`

```suggestion
const useCollectionData = <T, D>(COLLECTION_NAME: string) => {
```

File: src/modules/data/useData.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  const dataRef = useFirestore().collection(COLLECTION_NAME);
+  const data = useFirestoreCollectionData<T>(dataRef, {
+    idField: 'id',
+  });
+  const addData = async (d: D) => {
+    const creationDate = new Date();
+    return await dataRef.add({ ...d, creationDate });
+  };
+  const removeData = async (id: string) => {
```
Comment (jtaaa) [inline src/modules/data/useData.ts]: What do u think of `removeDoc` and `addDoc`?

**Review by jtaaa — APPROVED** [review]
LGTM

File: src/modules/data/useData.ts (hunk 2 above)
Comment (chriskrogh) [reply src/modules/data/useData.ts]: Makes sense to me

**Pull request merged**